Supply full URL to extract_links() and fix handling of relative URLs

feltcat opened the pull request:
Relating to #45

- Rewrite `fix_relative_urls` to allow passing a full URL as the base to compare against
- Adjust all functions that call `fix_relative_urls` to accept full URLs as well

This should fix things on the courlan side. There will still need to be changes to trafilatura to pass full URLs to the courlan functions in order to fix trafilatura's issues with relative URLs, but this change to courlan should still work independently of that.

adbar commented:
Hi @feltcat, good catch, the code is more stable with `urljoin`. I changed `urlparse`  to `urlsplit`, as we don't need the full parsing capacity it could be slightly faster.

Since we're at it do you see other cases we could add to the tests and also tackle in the code?

feltcat commented:
Added a few more tests. I don't know anything about URLs that start in `{`, but I added an example from another test. Not sure if that covers absolutely everything, but I'm struggling to think of any more cases off the top of my head. I used the existing tests as the guide for how the code should behave.

adbar commented:
@feltcat Thanks!
